pkg/api/v1/summary.go

```diff
@@ -131,3 +134,83 @@ func (s Server) ListTabSummariesHTTP(w http.ResponseWriter, r *http.Request) {
 
 	s.writeJSON(w, &resp)
 }
+
+// GetTabSummary returns the tab summary for the particular dashboard and tab.
+// Dashboard and tab names don't have to be normalized.
+// Returns an error iff
+// - dashboard or tab does not exist
+// - the server can't read summary from GCS bucket
+// - tab summary for particular tab doesn't exist
+func (s *Server) GetTabSummary(ctx context.Context, req *apipb.GetTabSummaryRequest) (*apipb.GetTabSummaryResponse, error) {
+	ctx, cancel := context.WithTimeout(ctx, s.Timeout)
+	defer cancel()
+
+	scope := req.GetScope()
+	cfg, err := s.getConfig(ctx, logrus.WithContext(ctx), scope)
+
+	// TODO(sultan-duisenbay): return canonical error codes
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch config from {%q}: %v", scope, err)
+	}
+
+	cfg.Mutex.RLock()
+	defer cfg.Mutex.RUnlock()
+
+	reqDashboardName, reqTabName := req.GetDashboard(), req.GetTab()
+
+	_, tabName, _, err := findDashboardTab(cfg, reqDashboardName, reqTabName)
+	if err != nil {
+		return nil, fmt.Errorf("invalid request input {%q, %q}: %v", reqDashboardName, reqTabName, err)
+	}
+
+	summary, err := s.fetchSummary(ctx, scope, reqDashboardName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch summary for dashboard {%q}: %v", reqDashboardName, err)
+	}
+
+	if summary == nil {
+		return nil, fmt.Errorf("summary for dashboard {%q} not found.", reqDashboardName)
+	}
+
+	// TODO(sultan-duisenbay): convert fractional part of timestamp to nanos of timestamppb
+	for _, tabSummary := range summary.GetTabSummaries() {
+		if tabSummary.DashboardTabName == tabName {
+			resp := apipb.GetTabSummaryResponse{
```

Comment:
nit: Should this be a utility function shared between this and the other function?

Reply:
Agree, done!